### PR TITLE
OCPBUGS-19378: [4.13] Backports of backup/restore fixes

### PR DESCRIPTION
**Pull request by tjungblu**

### CODE REVIEW

**Comment by tjungblu** [comment]
/hold

also waiting for https://github.com/openshift/cluster-etcd-operator/pull/1113 to land

**Comment by tjungblu** [comment]
seems there's some pod security issue in 4.13:
```
   	            	Original error:
        	            	pods "ip-10-0-134-12us-west-1computeinternal-debug" is forbidden: violates PodSecurity "restricted:latest": host namespaces (hostNetwork=true, hostPID=true, hostIPC=true), privileged (container "container-00" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "container-00" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "container-00" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volume "host" uses restricted volume type "hostPath"), runAsNonRoot != true (pod or container "container-00" must set securityContext.runAsNonRoot=true), runAsUser=0 (container "container-00" must not set runAsUser=0), seccompProfile (pod or container "container-00" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

File: test/e2e/backup_script_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 const (
 	masterNodeLabel = "node-role.kubernetes.io/master"
 	backupPath      = "/etc/kubernetes/backup-happy-path-2023-08-03_152313"
-	debugNamespace  = "default"
+	debugNamespace  = "openshift-etcd" // using a privileged ns to pass on fips
```
Comment (Elbehery) [inline test/e2e/backup_script_test.go:22]: lgtm 👍🏽

**Comment by tjungblu** [comment]
/lgtm 

(Mustafa already reviewed earlier)

**Pull request merged**